aws_lambda_powertools/event_handler/api_gateway.py

```diff
@@ -2386,6 +2387,7 @@ def register_route(func: Callable):
             methods = (method,) if isinstance(method, str) else tuple(method)
             frozen_responses = _FrozenDict(responses) if responses else None
             frozen_tags = frozenset(tags) if tags else None
+            frozen_security = _FrozenListDict(security) if security else None
```

Comment:
do we anticipate another non-hashable parameter we need to resolve now too?

I remember `tags` was one of them, now `security`. I dunno the implementation yet so I wonder if we need to stop and build a more robust solution OR if `security` is the last piece we overlooked

Reply:
At this moment, security is the last piece we overlooked. I can't say that we won't change this function or that we will never add another parameter, but at the moment we have the OpenAPI implementation almost complete, missing things like vendor extension and cookies, for example. But I don't think both will cause problems here.

Reply:
gotcha. Let's do this, if it happens with yet another parameter, we will stop and prioritize investigating a refactoring. The encoder is a good example of an area that we should've used a singledispatch to avoid a myriad of if/elif we overlooked at launch (issue with extremely large PRs).